src/gtk/toga_gtk/widgets/table.py

```diff
@@ -27,27 +27,36 @@ def create(self):
         self.native.set_min_content_height(200)
         self.native.interface = self.interface
 
+    # Gtk.TreeIter cannot be compared with __eq__ !!!
+    def compare_tree_iters(self, one, two):
+        return self.store.get_path(one) == self.store.get_path(two)
+
+    # find `tree_iter` in `self.rows`
+    def get_row(self, tree_iter):
+        if not isinstance(tree_iter, Gtk.TreeIter):
+            raise TypeError("expected Gtk.TreeIter, got {}".format(type(tree_iter)))
+
+        for row, it in self.rows.items():
+            if self.compare_tree_iters(tree_iter, it):
+                return row
+
+    def set_tree_iter(self, item, tree_iter):
+        item._tree_iters = getattr(item, "_tree_iters", {})
+        item._tree_iters[self] = tree_iter
+        self.rows[item] = tree_iter
+
     def _on_select(self, selection):
         if hasattr(self.interface, "_on_select") and self.interface.on_select:
-            tree_model, impl = selection.get_selected()
```

Comment:
`impl` is a pretty common convention in toga... I'm on the fence about changing it. 

On the one hand, this is closer to what gtk is doing. On the other hand, `impl` is how toga "thinks" about it.

Reply:
Good point. I'm pretty sure the main reason I changed it was because I was getting confused between all the different `impl`s going on in the same file, so I renamed it to what it actually represented to make the code easier to reason about. I'll mess with the names some more and perhaps come up with a reasonable compromise.